src/main/java/org/apache/maven/plugins/rar/internal/RarLifecycleMappingProvider.java

```diff
@@ -0,0 +1,105 @@
+package org.apache.maven.plugins.rar.internal;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.lifecycle.mapping.Lifecycle;
+import org.apache.maven.lifecycle.mapping.LifecycleMapping;
+
+/**
+ * {@code rar} packaging plugins bindings provider for {@code default} lifecycle.
+ */
+@Singleton
+@Named( "rar" )
+public final class RarLifecycleMappingProvider
+        implements Provider<LifecycleMapping>
+{
+    @SuppressWarnings( "checkstyle:linelength" )
+    private static final String[] BINDINGS =
+            {
+                    "process-resources", "org.apache.maven.plugins:maven-resources-plugin:3.2.0:resources",
```

Comment:
Shouldn't the goal spec rather leave out the version?
I don't think the plugin should bind to any specific version here.
AFAIK the format `<groupId>:<artifactId>:<goal>` works just as well: https://github.com/apache/maven/blob/8e5676c84a782d51c4156be059b6ce9a3dc3e1e1/maven-core/src/main/java/org/apache/maven/lifecycle/internal/DefaultLifecyclePluginAnalyzer.java#L242

Reply:
All the bindings are done against some specific plugin version, just see maven core https://github.com/apache/maven/blob/maven-3.9.x/maven-core/src/main/resources/META-INF/plexus/default-bindings.xml

This serves the purpose that maven defines "own" level of plugins, that users CAN override (and should) by locking down plugin versions in POM. If we'd not lock down version, maven could radically change it's way of working as it could pick up even MAJOR plugin version change. Hence, maven locks down in "mild" way (if user does not locks down on it's own), ensuring that maven version X will always use same set of plugins (unless overridden in POM).

Reply:
> Hence, maven locks down in "mild" way 

I understand for Maven Core, but not for lifecycle bindings contained in plugins. Which binding takes precedence by the way? Which version wins? IMHO only Maven Core should lock down versions in a mild way, but not plugins themselves....

Reply:
POM wins, if user did not set version in POM, then lifecycle wins (as sort of "defaults").

Reply:
Ah, you meant lifecycle not plugin versions within lifecycle.... build ext http://takari.io/book/91-maven-classloading.html

But plugin lifecycle "replaces" maven core lifecycle, there is no "merge" of those (plugin+core), hence, plugin will completely replace core one, that is again okay, as in this case, rar plugin version  may be set by user....

Reply:
That means e.g for Maven 3.8.6 when not defining explicit plugin versions in the pom.xml :
The versions of the plugins used in the default binding for packaging `rar`:

1. Are defined by Maven Core if this plugin is not loaded with extensions classloader
2. Are defined by this plugin if it is loaded with extensions classloader (which somehow was not considered production-ready in https://github.com/apache/maven-apache-parent/pull/92)

Reply:
How this works if users
 uses this plugin but wants to use never version of compiler plugin?

Reply:
pom.xml always wins IIUC: https://github.com/apache/maven/blob/8e5676c84a782d51c4156be059b6ce9a3dc3e1e1/maven-core/src/main/java/org/apache/maven/lifecycle/internal/DefaultLifecyclePluginAnalyzer.java#L201